account/templates/account/dashboard.html

```diff
@@ -7,25 +7,263 @@
+<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
+    <script src="https://cdn.jsdelivr.net/npm/popper.js@1.12.9/dist/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.0.0/dist/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
+
 <script src="{% static 'js/call.js' %}"></script>
 <script>
 
 
+let localStream;
+let peerConnection;
+let otherUser;
+let remoteRTCMessage;
+let localVideo = document.querySelector('#localVideo');
+
+let callInProgress = false;
+let iceCandidatesFromCaller = [];
+const constraints={
+    'video': true,
+    'audio': false, 
+}
 var callSocket = new WebSocket(
         'ws://'
         + window.location.host
         + '/ws/call/'
     );
 
-function beReady(){
-    console.log('I am ready')
+
+
+let pcConfig = {
+    "iceServers":
+        [
+            { "url": "stun:stun.jap.bloggernepal.com:5349" },
+            {
+                "url": "turn:turn.jap.bloggernepal.com:5349",
+                "username": "guest",
+                "credential": "somepassword"
+            },
+            {"url": "stun:stun.l.google.com:19302"}
+        ]
+};
+
+
+function handleRemoteStreamAdded(event) {
+    console.log('Remote stream added.');
+    remoteStream = event.stream;
+    remoteVideo.srcObject = remoteStream;
+}
+
+function handleRemoteStreamRemoved(event) {
+    console.log('Remote stream removed. Event: ', event);
+    remoteVideo.srcObject = null;
+    localVideo.srcObject = null;
+}
+
+
+function sendICEcandidate(data) {
+    //send only if we have caller, else no need to
+    console.log("Send ICE candidate");
+    // socket.emit("ICEcandidate", data)
+    callSocket.send(JSON.stringify({
+        type: 'ICEcandidate',
+        data
+    }));
+
+}
+
+
+function handleIceCandidate(event) {
+    // console.log('icecandidate event: ', event);
+    if (event.candidate) {
+        console.log("Local ICE candidate");
+        // console.log(event.candidate.candidate);
+
+        sendICEcandidate({
+            user: otherUser,
+            rtcMessage: {
+                label: event.candidate.sdpMLineIndex,
+                id: event.candidate.sdpMid,
+                candidate: event.candidate.candidate
+            }
+        })
+
+    } else {
+        console.log('End of candidates.');
+    }
+}
+
+
+function createPeerConnection() {
+    try {
+        peerConnection = new RTCPeerConnection(pcConfig);
+        //peerConnection = new RTCPeerConnection();
+        peerConnection.onicecandidate = handleIceCandidate;
+        peerConnection.onaddstream = handleRemoteStreamAdded;
```

Comment:
```suggestion
        peerConnection.ontrack = handleRemoteStreamAdded;
```

Reply:
ontrack is not working in our scenario, but onaddstream is working perfectly. While using ontrack remoteStream is not working

Reply:
I tried this out and i got it working. Did you check that the handleRemoteStreamAdded function also had the suggested changes?

Reply:
It is working now.